src/lib/middleware/api-token-middleware.ts

```diff
@@ -11,11 +11,13 @@ const isProxyApi = ({ path }) => {
         return;
     }
 
+    // Handle all our current proxy paths which will redirect to the new
+    // embedded proxy endpoint
     return (
-        (!path.startsWith('/api/client') &&
-            !path.startsWith('/api/admin') &&
-            path.includes('proxy')) ||
-        (path && path.startsWith('/api/frontend'))
+        path.startsWith('/api/default/proxy') ||
+        path.startsWith('/api/development/proxy') ||
+        path.startsWith('/api/production/proxy') ||
```

Comment:
Nice! Can there be custom environment names in this case, or will it always be one of these three?

Reply:
I'm double checking this, but according to our infra as code, those are the only ones we use.